mono/sgen/sgen-alloc.c

```diff
@@ -507,16 +514,62 @@ sgen_alloc_obj_mature (GCVTable vtable, size_t size)
 void
 sgen_clear_tlabs (void)
 {
+	guint64 total_bytes_allocated_globally = 0;
+
 	FOREACH_THREAD_ALL (info) {
 		/* A new TLAB will be allocated when the thread does its first allocation */
 		info->total_bytes_allocated += info->tlab_next - info->tlab_start;
+		total_bytes_allocated_globally += info->total_bytes_allocated;
 		info->tlab_start = NULL;
 		info->tlab_next = NULL;
 		info->tlab_temp_end = NULL;
 		info->tlab_real_end = NULL;
 	} FOREACH_THREAD_END
+
+	sgen_set_bytes_allocated_attached (total_bytes_allocated_globally);
+} 
+
+void sgen_update_allocation_count (void)
+{
+	guint64 total_bytes_allocated_globally = 0;
+
+	FOREACH_THREAD_ALL (info) {
+		total_bytes_allocated_globally += info->tlab_next - info->tlab_start;
+		total_bytes_allocated_globally += info->total_bytes_allocated;
+	} FOREACH_THREAD_END
+
+	sgen_set_bytes_allocated_attached (total_bytes_allocated_globally);
 }
 
+void
+sgen_set_bytes_allocated_attached (guint64 bytes)
+{
+	bytes_allocated_attached = bytes;
+}
+
+void
+increment_bytes_allocated_detached(guint64 bytes) 
```

Comment:
space here. We have the general convention to have prefixes (like sgen) for methods that are not static.